lib/src/builder/generation/declaration_parsing.dart

```diff
@@ -477,18 +537,21 @@ class ParsedDeclarations {
   final bool declaresComponent;
 
   /// Helper function that returns the single value of a [list], or null if it is empty.
-  static dynamic singleOrNull(List list) => list.isNotEmpty ? list.single : null;
+  static singleOrNull(List list) => list.isNotEmpty ? list.single : null;
```

Comment:
#nit Ooh we could make this generic (generics functions weren't supported when this method was authored):
```suggestion
  static T singleOrNull<T>(List<T> list) => list.isNotEmpty ? list.single : null;
```